pkg/machine/qemu/machine_test.go

```diff
@@ -36,5 +43,13 @@ func TestPropagateHostEnv(t *testing.T) {
 	tokens := strings.Split(cmdLine[1], ",string=")
 	decodeString, err := base64.StdEncoding.DecodeString(tokens[1])
 	assert.NoError(t, err)
-	assert.Equal(t, fmt.Sprintf("HTTP_PROXY=\"proxy\"|SSL_CERT_FILE=\"%s/foo.cert\"|SSL_CERT_DIR=%q", machine.UserCertsTargetPath, machine.UserCertsTargetPath), string(decodeString))
+
+	assert.Contains(t, string(decodeString), "http_proxy=\"host.containers.internal:8888\"")
+	assert.Contains(t, string(decodeString), "https_proxy=\"host.containers.internal:8888\"")
+	assert.Contains(t, string(decodeString), "ftp_proxy=\"domain.com:8888\"")
+	assert.Contains(t, string(decodeString), "HTTP_PROXY=\"proxy\"")
+	assert.Contains(t, string(decodeString), "FTP_PROXY=\"proxy\"")
+	assert.Contains(t, string(decodeString), "NO_PROXY=\"localaddress\"")
+	assert.Contains(t, string(decodeString), fmt.Sprintf("SSL_CERT_FILE=\"%s/foo.cert\"", machine.UserCertsTargetPath))
+	assert.Contains(t, string(decodeString), fmt.Sprintf("SSL_CERT_DIR=%q", machine.UserCertsTargetPath))
```

Comment:
The previous approach could cause unit testing errors in certain situations because it relied on the order of maps, while maps in Golang do not guarantee the order of keys.

Reply:
I think this would benefit from a table based test, i.e. create a slice with all expected strings and then just loop over it:
```go
for _, expect := range variables  {
		assert.Contains(t, string(decodeString), expected)
}
```

Reply:
Good idea 👍. Modification completed.